packages/auth/amplify_auth_cognito_dart/test/plugin/auth_providers_test.dart

```diff
@@ -47,66 +53,97 @@ void main() {
-      final authProvider = CognitoIamAuthProvider();
-      final authorizedRequest = await authProvider.authorizeRequest(
-        _generateTestRequest(),
-        options: const IamAuthProviderOptions(
-          region: 'us-east-1',
-          service: AWSService.appSync,
-        ),
-      );
-      // Note: not intended to be complete test of sigv4 algorithm.
-      expect(authorizedRequest.headers[AWSHeaders.authorization], isNotEmpty);
-      const userAgentHeader =
-          zIsWeb ? AWSHeaders.amzUserAgent : AWSHeaders.userAgent;
-      expect(
-        authorizedRequest.headers[AWSHeaders.host],
-        isNotEmpty,
-        skip: zIsWeb,
-      );
-      expect(
-        authorizedRequest.headers[userAgentHeader],
-        contains('aws-sigv4'),
-      );
+      test('signs a request when calling authorizeRequest', () async {
+        final authProvider = CognitoIamAuthProvider();
+        final authorizedRequest = await authProvider.authorizeRequest(
+          _generateTestRequest(),
+          options: const IamAuthProviderOptions(
+            region: 'us-east-1',
+            service: AWSService.appSync,
+          ),
+        );
+        // Note: not intended to be complete test of sigv4 algorithm.
+        expect(authorizedRequest.headers[AWSHeaders.authorization], isNotEmpty);
+        const userAgentHeader =
+            zIsWeb ? AWSHeaders.amzUserAgent : AWSHeaders.userAgent;
+        expect(
+          authorizedRequest.headers[AWSHeaders.host],
+          isNotEmpty,
+          skip: zIsWeb,
+        );
+        expect(
+          authorizedRequest.headers[userAgentHeader],
+          contains('aws-sigv4'),
+        );
+      });
+
+      test('throws when no options provided', () async {
+        final authProvider = CognitoIamAuthProvider();
+        await expectLater(
+          authProvider.authorizeRequest(_generateTestRequest()),
+          throwsA(isA<AuthException>()),
+        );
+      });
     });
 
-    test('throws when no options provided', () async {
-      final authProvider = CognitoIamAuthProvider();
-      await expectLater(
-        authProvider.authorizeRequest(_generateTestRequest()),
-        throwsA(isA<AuthException>()),
-      );
+    group('CognitoUserPoolsAuthProvider', () {
+      test('gets raw access token from Amplify.Auth.fetchAuthSession',
+          () async {
+        final authProvider = CognitoUserPoolsAuthProvider();
+        final token = await authProvider.getLatestAuthToken();
+        expect(token, accessToken.raw);
+      });
+
+      test('adds access token to header when calling authorizeRequest',
+          () async {
+        final authProvider = CognitoUserPoolsAuthProvider();
+        final authorizedRequest = await authProvider.authorizeRequest(
+          _generateTestRequest(),
+        );
+        expect(
+          authorizedRequest.headers[AWSHeaders.authorization],
+          accessToken.raw,
+        );
+      });
```

Comment:
good suggestion, beefed up the tests per this suggestion

Reply:
Another good one will be to test the auth provider repo itself at some point (not sure if you've already done this). Basically register two mock plugins - in one of them register an auth provider, and in the other try to get it, and try to see how and why this could fail.

Reply:
Thanks, there is already similar test in core for auth provider repo where it's defined.